### PR TITLE
feat(limits): Ignore prNotPendingHours when stabilityDays is set

### LINKED ISSUE #7449: Ignore prNotPendingHours setting if stabilityDays is configured
**What would you like Renovate to be able to do?**

Dynamically ignore prNotPendingHours for a branch if stabilityDays is set

**Did you already have any implementation ideas?**

Hopefully this can be a simple logic check at the time of creation

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**

Massing the value of prNotPendingHours, but I think it's easier to just ignore it

**Is this a feature you'd be interested in implementing yourself?**

No

### PR DESCRIPTION
**Pull request by zharinov**
Closes #7449

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
No new tests? 🤔

**Comment by zharinov** [comment]
Yes, I also need to test it manually

**Comment by zharinov** [comment]
Added tests look pretty straighforward, to the another hand I'm not sure it can be tested manually today :)

File: lib/workers/pr/index.spec.ts [hunk 1]
```diff
@@ -361,6 +362,17 @@ describe('workers/pr', () => {
...
+      platform.getBranchStatus.mockResolvedValueOnce(BranchStatus.yellow);
+      git.getBranchLastCommitTime.mockImplementationOnce(() =>
+        Promise.resolve(new Date())
+      );
+      config.prCreation = 'not-pending';
+      config.stabilityDays = 42;
+      const { prResult, pr } = await prWorker.ensurePr(config);
+      expect(prResult).toEqual(PrResult.Created);
+      expect(pr).toBeDefined();
+    });
```
Comment (zharinov) [inline lib/workers/pr/index.spec.ts:375]: Compare with tests above and below

File: lib/workers/pr/index.spec.ts [hunk 2]
```diff
@@ -570,6 +582,16 @@ describe('workers/pr', () => {
...
+    it('should return no PR if stabilityDays set', async () => {
+      config.automerge = true;
+      config.automergeType = 'branch';
+      config.stabilityDays = 42;
+      platform.getBranchStatus.mockResolvedValueOnce(BranchStatus.yellow);
+      git.getBranchLastCommitTime.mockResolvedValueOnce(new Date('2018-01-01'));
+      const { prResult, pr } = await prWorker.ensurePr(config);
+      expect(prResult).toEqual(PrResult.BlockedByBranchAutomerge);
+      expect(pr).toBeUndefined();
+    });
```
Comment (zharinov) [inline lib/workers/pr/index.spec.ts:594]: Compare with tests above and below

File: lib/workers/pr/index.ts [hunk 3]
```diff
@@ -143,7 +143,10 @@ export async function ensurePr(
     logger.debug(
       `Branch is configured for branch automerge, branch status) is: ${await getBranchStatus()}`
     );
-    if ((await getBranchStatus()) === BranchStatus.yellow) {
+    if (
+      !config.stabilityDays &&
```
Comment (viceice) [inline lib/workers/pr/index.ts:147]: what value whill `stabilityDays` when reached? Will stabilityDays  always suppress  this check?

Comment (rarkins) [reply lib/workers/pr/index.ts:147]: I believe it will still be the same value. If the branch is still pending due to some *other* check then it will remain that way forever.

File: lib/workers/pr/index.ts [hunk 4]
```diff
@@ -194,10 +201,7 @@ export async function ensurePr(
       const elapsedHours = Math.round(
         (currentTime.getTime() - lastCommitTime.getTime()) / millisecondsPerHour
       );
-      if (
-        !dependencyDashboardCheck &&
-        elapsedHours < config.prNotPendingHours
-      ) {
+      if (elapsedHours < config.prNotPendingHours) {
```
Comment (rarkins) [inline lib/workers/pr/index.ts:204]: I find this rearrangement of logic a little hard to follow. What was the reasoning?

Comment (zharinov) [reply lib/workers/pr/index.ts:204]: I wanted to make `Branch is ${elapsedHours} hours old - skipping PR creation` log message appear only when `prNotPendingHours` is in game.

Comment (zharinov) [reply lib/workers/pr/index.ts:204]: More broadly, it would be nice to have log messages for each reason, but maybe with separte PR

**Review by rarkins — CHANGES_REQUESTED** [review]
One concern: if the user configures stabilityDays but there's no releaseTimestatmp, then the branch will be pending forever. Is it possible instead to use the logic `if (config.stabilityStatus !== BranchStatus.yellow)` ? i.e. instead of skipping prNotPendingHours if stabilityDays is configured then instead skip it as long as stabilityDays is still yellow?

**Comment by rarkins** [comment]
@zharinov I have refactored to use `config.stabilityStatus` in workers/pr. Can you verify it's working as we want and update tests?
